packages/regl-worldview/src/commands/GLText.js

```diff
@@ -272,63 +299,100 @@ const vert = `
+    vec2 charOffsetCoord = vec2(0.5) - ((backgroundDestSize - srcSize) / 2. + charOffset) / backgroundDestSize;
+    vec2 textAtlasCoord = (charOffset + srcOffset + texCoord * backgroundDestSize) / atlasSize;
+
+    // Compute the coords for the full line
+    vec2 lineCoord = (texCoord * backgroundDestSize + destOffset) / roundedBackgroundSize;
+    vLineBackgroundCoord = vec2(lineCoord.x, texCoord.y) * 2. - 1.; // Transform from [0,1] to [-1,1]
+
+    vCharToBackgroundSizeRatio = srcSize / backgroundDestSize;
+    vLineBackgroundRadiusRatio = roundedBackgroundSize.y / roundedBackgroundSize.x;
+    
     vBackgroundColor = backgroundColor;
-    vHighlightColor = highlightColor;
-    vEnableHighlight = enableHighlight;
     vBillboard = billboard;
+    vCharOffset = charOffsetCoord;
+    vCharBackgroundCoord = texCoord;
+    vTextColor = textColor;
+    vTextAtlasCoord = textAtlasCoord;
   }
 `;
 
 const frag = `
   #extension GL_OES_standard_derivatives : enable
   precision mediump float;
-  uniform mat4 projection;
+
   uniform sampler2D atlas;
   uniform float cutoff;
+  uniform bool isHitmap;
   uniform bool scaleInvariant;
   uniform float scaleInvariantSize;
-  uniform bool isHitmap;
-
-  varying vec2 vTexCoord;
-  varying float vEnableBackground;
-  varying vec4 vForegroundColor;
-  varying vec4 vBackgroundColor;
-  varying vec4 vHighlightColor;
-  varying float vEnableHighlight;
+  uniform float borderRadiusSize;
+  
+  varying vec2 vLineSize;
+
+  varying float vLineBackgroundRadiusRatio;
+  varying vec2 vCharBackgroundCoord;
+  varying vec2 vTextAtlasCoord;
+  varying vec4 vTextColor;
+  varying vec2 vCharOffset;
+  varying vec2 vCharToBackgroundSizeRatio;
   varying float vBillboard;
+  varying vec2 vLineBackgroundCoord;
+  varying vec4 vBackgroundColor;
 
-  void main() {
-    float dist = texture2D(atlas, vTexCoord).a;
+  // fwidth(dist) is used to provide some anti-aliasing. However it's currently only used
+  // when the solid background is enabled, because the alpha blending and
+  // depth test don't work together nicely for partially-transparent pixels.
+  float getAntialisedStep(float dist, float cutoff) {
+    return smoothstep(1. - cutoff - fwidth(dist), 1. - cutoff, dist);
+  }
 
-    // fwidth(dist) is used to provide some anti-aliasing. However it's currently only used
-    // when the solid background is enabled, because the alpha blending and
-    // depth test don't work together nicely for partially-transparent pixels.
-    float edgeStep = smoothstep(1.0 - cutoff - fwidth(dist), 1.0 - cutoff, dist);
+  float rectangleMask(vec2 uv, vec2 position, vec2 size) {
+    float t = 0.;
+    if ((uv.x > position.x - size.x / 2.) && (uv.x < position.x + size.x / 2.)
+      && (uv.y > position.y - size.y / 2.) && (uv.y < position.y + size.y / 2.)) {
+      t = 1.;
+    }
+    return t;
+  }
+
+  float roundedRectangleMask(vec2 position, vec2 size, float radius) {
```

Comment:
These could probably be the same function since they do the same thing, but one has rounded corners

Reply:
I guess `rectangleMask` can call `roundedRectangleMask` with no radius, maybe?